classes/submission/PKPSubmissionMetadataFormImplementation.inc.php

```diff
@@ -139,6 +140,9 @@ function execute($submission, $request) {
 		$submission->setSource($this->_parentForm->getData('source'), null); // Localized
 		$submission->setCitations($this->_parentForm->getData('citations'));
 
+		// Update submission locale
+		$submission->setLocale($this->_parentForm->getData('locale'));
```

Comment:
It looks like this value is currently trusted without validation -- it needs to be one of the supported locales.